File: Kevin-Muthiani/Arrays_and_Strings/reverse_array_in_place.py
# Write a function reverseArray(A) that takes in an array A and reverses it, 
# without using another array or collection data structure i.e. in-place.

def reverseArray(arr):
    n = len(arr)
    
    for i in range(n//2):
      arr[i], arr[n-1-i] = arr[n-1-i], arr[i]


# Test function
if __name__ == "__main__":
  myArray = [10, 5, 7, 6, 9]
  reverseArray(myArray)
  print(myArray)

File: AugustineKadima/Arrays_And_Strings/reverseArrayInPlace.js


const reverseArray = (A) => {
    let n = A.length - 1;
    let i;
   
    for(i = 0; i < n; i++){
        let temp = A[n];
        if(i < n){
            A[n] = A[i];
            A[i] = temp;
            n--;
        }
    } 
    console.log(A);

}
reverseArray([9, 6, 5, 10, 7, 4, 3, 4, 5]);